templates/cluster-template.yaml

```diff
@@ -1,107 +1,161 @@
+  nodeRefs:
+    - name: mynode
+      secretRef:
+        name: "${CLUSTER_NAME}"
+  storage:
+    name: "${CLUSTER_NAME}"
+    path: ""
+
+---
+
+apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+kind: KubeadmControlPlane
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  kubeadmConfigSpec:
+    clusterConfiguration:
+      etcd:
+        local:
+          dataDir: /var/lib/etcd
+      networking:
+        dnsDomain: cluster.local
+        serviceSubnet: 10.96.0.0/12
+        podSubnet: 10.98.0.0/16
+    initConfiguration:
+      localAPIEndpoint:
+        advertiseAddress: "${CONTROLPLANE_HOST}"
+        bindPort: 6443
+      nodeRegistration:
+        criSocket: unix:///var/run/containerd/containerd.sock
+    postKubeadmCommands:
+    - "curl -L https://dl.k8s.io/release/v1.27.3/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl"
+    - "chmod +x /usr/local/bin/kubectl"
+    - "reboot now"
+    preKubeadmCommands: []
+  machineTemplate:
+    infrastructureRef:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: ProxmoxMachineTemplate
+      name: ${CLUSTER_NAME}-controlplane
+  replicas: 1
+  version: v1.26.5
+
+---
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: ProxmoxMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-controlplane
+spec:
+  template:
+    spec:
+      image:
+        url: https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64-disk-kvm.img
+      hardware:
+        cpu: 4
+        memory: 8192
+      network:
+        ipConfig:
+          gateway: ${GATEWAY_ADDRESS}
+          ip: "${CONTROLPLANE_HOST}/32"
+        nameServer: ${NAMESERVER_ADDRESS}
+
+---
+
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  clusterName: "${CLUSTER_NAME}"
+  replicas: 1
+  selector:
+    matchLabels: null
+  template:
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfigTemplate
+          name: worker
+      clusterName: "${CLUSTER_NAME}"
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: ProxmoxMachineTemplate
+        name: ${CLUSTER_NAME}-worker
+      version: v1.26.5
+
+---
+
+apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+kind: KubeadmConfigTemplate
+metadata:
+  name: worker
+spec:
+  template:
+    spec:
+      postKubeadmCommands:
+        - "curl -L https://dl.k8s.io/release/v1.27.3/bin/linux/amd64/kubectl -o /usr/local/bin/kubectl"
+        - "chmod +x /usr/local/bin/kubectl"
+        - "reboot now"
+
+---
+
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: ProxmoxMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-worker
+spec:
+  template:
+    spec:
+      image:
+        url: https://cloud-images.ubuntu.com/jammy/current/jammy-server-cloudimg-amd64-disk-kvm.img
+      hardware:
+        cpu: 4
+        memory: 4096
+
+---
+
+apiVersion: v1
+stringData:
```

Comment:
@sp-yduck I would like to suggest stringData in favor of data because its then easier if you have a clusterctl-config file. (s. https://cluster-api.sigs.k8s.io/clusterctl/configuration.html) Also the envs this would look "cleaner".

Reply:
oh thank you for the suggestion but unfortunately it needs some code change around [here](https://github.com/sp-yduck/cluster-api-provider-proxmox/blob/main/cloud/scope/clients.go#L47-L51) and if we accept both `data` and `stringData` it will become slight complecated. 
since I can see [official doc](https://cluster-api.sigs.k8s.io/clusterctl/configuration.html#variables) is also using base64 encoded values. so let us keep as it is.

Reply:
Are you sure code changes are necessary? stringData gets converted into base64 data in the cluster as far as I know. I also could use this string data to bootstrap a cluster. (This was the first thing I changed on the first try)

Reply:
Ahh but the docs, I see... But not necessary imo.